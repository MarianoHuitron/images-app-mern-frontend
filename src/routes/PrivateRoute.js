import React from 'react'
import { Redirect, Route } from 'react-router-dom';
import { Navbar } from '../components/navbar/Navbar';

export const PrivateRoute = ({ isAuth, component: Component, ...rest }) => {
  
    return (
        <>
            <Navbar />
             <Route 
                {...rest} 
                component={ (props) => (
                    (isAuth)
                    ? <Component { ...props } />
                    : <Redirect to="/login" />
                )}   
            />
        </>

       
    );
    
}
