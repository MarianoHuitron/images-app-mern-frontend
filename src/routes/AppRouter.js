import React, { useEffect } from 'react'
import { useDispatch, useSelector } from 'react-redux';
import {
    BrowserRouter as Router,
    Switch,
    Redirect
  } from "react-router-dom";
import { startChecking } from '../actions/auth';
import { Loading } from '../components/loading/Loading';
import { HomePage } from '../pages/home/HomePage';
import { LoginPage } from '../pages/login/LoginPage';
import { RegisterPage } from '../pages/register/RegisterPage';
import { UploadPage } from '../pages/upload/UploadPage';
import { PrivateRoute } from './PrivateRoute';
import { PublicRoute } from './PublicRoute';

export const AppRouter = () => {


    const dispatch = useDispatch();

    const { checking, uid } = useSelector(state => state.auth);

    useEffect(() => {
        dispatch( startChecking() );
    }, [dispatch]);

    if(checking) {
        return (<Loading />);
    }

    

    return (
        <Router>
            <div>
                <Switch>
                    
                    <PublicRoute
                        exct
                        path="/login"
                        component={ LoginPage }
                        isAuth={ !!uid }
                    />
                    <PublicRoute
                        exct
                        path="/register"
                        component={ RegisterPage }
                        isAuth={ !!uid }
                    />

                    <PrivateRoute
                        exact
                        path="/"
                        component={ HomePage }
                        isAuth={ !!uid }
                    />
                    <PrivateRoute
                        exact
                        path="/upload"
                        component={ UploadPage }
                        isAuth={ !!uid }
                    />

                    <Redirect to="/" />

                </Switch>
            </div>
        </Router>
    )
}
