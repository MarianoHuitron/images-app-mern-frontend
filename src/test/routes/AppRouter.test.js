import React from 'react';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';

import configureStore from 'redux-mock-store'
import thunk from 'redux-thunk'

import '@testing-library/jest-dom';
import { AppRouter } from '../../routes/AppRouter';

const middlewares = [thunk];
const mockstore = configureStore(middlewares);

describe('Testing <AppRouter />', () => {
    
    test('loading...', () => {
        
        const initState = {
            auth: {
                checking: true
            }
        };
        const store = mockstore(initState);

        const wrapper = mount(
            <Provider store={ store } >
                <AppRouter />
            </Provider>
        )
        
        expect( wrapper ).toMatchSnapshot();

    });

    test('Public routes', () => {
        const initState = {
            auth: {
                checking: false
            },
            ui: {
                errorForm: {}      
            }
        };
        const store = mockstore(initState);

        const wrapper = mount(
            <Provider store={ store } >
                <AppRouter />
            </Provider>
        )
        
        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('form').exists() ).toBe(true);
    });

    test('Private routes', () => {
        const initState = {
            auth: {
                checking: false,
                uid: '123',
                name: 'Jose'
            },
            ui: {
                errorForm: {}      
            }
        };
        const store = mockstore(initState);

        const wrapper = mount(
            <Provider store={ store } >
                <AppRouter />
            </Provider>
        )
        
        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('img').exists() ).toBe(true);
    })
    
    

})
