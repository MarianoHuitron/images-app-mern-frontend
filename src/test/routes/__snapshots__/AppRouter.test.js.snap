// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <AppRouter /> Private routes 1`] = `
<div>
  <div
    className="w-full  flex"
  >
    <div
      className="mx-auto pt-12 animate__animated animate__fadeIn"
    >
      <img
        alt="welcome"
        className="text-center h-56"
        src="./assets/welcome.svg"
      />
      <div
        className="text-center mt-10"
      >
        <button
          className="py-3 bg-green-500 text-white w-full hover:bg-green-600 cursor-pointer rounded outline-none"
          onClick={[Function]}
        >
          Logout
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`Testing <AppRouter /> Public routes 1`] = `
<div>
  <div
    className="w-full h-screen flex bg-gray-400 "
  >
    <div
      className="container mx-auto p-2 my-24"
    >
      <div
        className="max-w-sm mx-auto bg-white px-5 py-10 rounded shadow-xl animate__animated animate__fadeInDown"
      >
        <div
          className="text-center mb-8"
        >
          <h1
            className="font-bold text-2xl"
          >
            Welcome to 
            <span
              className="text-green-500"
            >
               Login App 
            </span>
             
          </h1>
        </div>
        <form
          onSubmit={[Function]}
        >
          <div
            className="mt-5"
          >
            <label>
              Email
            </label>
            <input
              autoComplete="off"
              className="block w-full p-2 border rounded border-gray-500 outline-none"
              name="email"
              placeholder="example@emal.com"
              type="email"
            />
            <p
              className="text-red-500"
            />
          </div>
          <div
            className="mt-5"
          >
            <label>
              Password
            </label>
            <input
              className="block w-full p-2 border rounded border-gray-500 outline-none"
              name="password"
              placeholder="********"
              type="password"
            />
            <p
              className="text-red-500"
            />
          </div>
          <div
            className="mt-10 text-center"
          >
            <button
              className="py-3 bg-green-500 text-white w-full hover:bg-green-600 cursor-pointer rounded outline-none"
            >
              Login
            </button>
          </div>
          <div
            className="mt-5"
          >
            <p
              className="text-gray-600"
            >
              Don't have an account?
              <a
                className="text-green-400"
                href="/register"
                onClick={[Function]}
              >
                 Sign up here 
              </a>
            </p>
          </div>
        </form>
      </div>
    </div>
  </div>
</div>
`;

exports[`Testing <AppRouter /> loading... 1`] = `
<div
  className="container-loader"
>
  <div
    className="grid"
  >
    <div
      className="box box-1"
    />
    <div
      className="box box-2"
    />
    <div
      className="box box-3"
    />
    <div
      className="box box-4"
    />
    <div
      className="box box-5"
    />
    <div
      className="box box-6"
    />
    <div
      className="box box-7"
    />
    <div
      className="box box-8"
    />
    <div
      className="box box-9"
    />
  </div>
</div>
`;
